reed-solomon-novelpoly/src/novel_poly_basis/mod.rs

```diff
@@ -231,6 +233,52 @@ impl ReedSolomon {
 
 		Ok(acc)
 	}
+
+	/// Reconstruct from the set of systematic chunks.
+	/// Systematic chunks are the first `k` chunks, which contain the initial data.
+	///
+	/// Provide a vector containing chunk data. If too few chunks are provided, recovery is not
+	/// possible.
+	/// The result may be padded with zeros. Truncate the output to the expected byte length.
+	pub fn reconstruct_from_systematic<S: Shard>(&self, chunks: Vec<S>) -> Result<Vec<u8>> {
+		let Some(first_shard) = chunks.first() else {
+			return Err(Error::NeedMoreShards { have: 0, min: self.k, all: self.n });
+		};
+		let shard_len = AsRef::<[[u8; 2]]>::as_ref(first_shard).len();
+
+		if shard_len == 0 {
+			return Err(Error::InconsistentShardLengths { first: 0, other: 0 });
+		}
```

Comment:
why is this checked needed? we don't seem to check it in regular reconstruct so I assume it's implied somehow, also the returned error doesn't match the intent exactly

Reply:
I added an `EmptyShard` error